collector/bonding_linux.go

```diff
@@ -31,6 +32,7 @@ type bondingCollector struct {
 
 func init() {
 	Factories["bonding"] = NewBondingCollector
+	CollectorsEnabledState["bonding"] = flag.Bool("collectors.bonding.enabled", false, "enables bonding-collector")
```

Comment:
If you define the flags in the arch files (or files with arch build tags) they should only get compiled there.